### DISCUSSION TITLE
[Bug] setOptions url doesn't work

### DISCUSSION THREAD

**Discussion by cathino** [katspaugh/wavesurfer.js#3177] [Q&A]
## Bug description
When I call `wavesurfer.setOptions({ url: newAudioBase64String });`, it dowsn't work.
`newAudioBase64String ` is a base64 string of e new audio file.

## Environment
 - Browser: Chrome and Firefox
 - version: 7.2.2

## Minimal code snippet

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <div id="waveform"></div>

    <script src="https://unpkg.com/wavesurfer.js@7"></script>
    <script>
      const song0 = "data:audio/mpeg;base64,SUDfgew...";

      const song1 = "data:audio/mpeg;base64,SUQzB...";

      const wavesurfer = WaveSurfer.create({
        container: "#waveform",
        waveColor: "#4F4A85",
        progressColor: "#383351",
        url: song0,
      });

      wavesurfer.once("interaction", () => {
        wavesurfer.play();
      });

      setTimeout(() => {
        wavesurfer.setOptions({ url: song1 }); // Not updated as expected here.

        console.log("changed");
      }, 2000);
    </script>
  </body>
</html>

```

## Expected result
Update with new URL.

**Comment by katspaugh** [comment]
You should call [`load`](https://wavesurfer-js.org/docs/classes/wavesurfer.default#load) instead of `setOptions` to change the URL.

**Comment by cathino** [comment]
Actually, at first, I used `load` in my vue3 project, but it caused an error `DOMException: The element has no supported sources.`.  I tried setting `url` or `media` in the options, and `setOptions url`/`setOptions media/`loadBlob`/`load` when I updated the audio.
But in a single HTML file, it works. Maybe it's about vue3.

**Comment by katspaugh** [comment]
I don't much about Vue, but [some of the discussions about it](https://github.com/katspaugh/wavesurfer.js/discussions?discussions_q=vue) mentioned some issues:
> reactive or ref cannot be used to declare instances, as this can easily lead to such issues.
